cmd/credentialdiggerScan.go

```diff
@@ -0,0 +1,241 @@
+		// Download custom rule file from this URL
+		log.Entry().Debugf("Download custom ruleset from %v", config.RulesDownloadURL)
+		dlClient := piperhttp.Client{}
+		ruleFile := filepath.Join(cdHome, "backend", "custom-rules.yml")
+		dlClient.DownloadFile(config.RulesDownloadURL, ruleFile, nil, nil)
+	} else {
+		log.Entry().Debug("Use standard ruleset")
+	}
+	cmd_list := []string{"add_rules", "--sqlite", piperDbName, ruleFile}
+	return executeCredentialDiggerProcess(service, cmd_list)
+}
+
+func credentialdiggerGetDiscoveries(config *credentialdiggerScanOptions, telemetryData *telemetry.CustomData, service credentialdiggerUtils) error {
+	log.Entry().Info("Get discoveries")
+	cmd_list := []string{"get_discoveries", config.Repository, "--sqlite", piperDbName,
+		"--save", piperReportName}
+	// Export all the discoveries or export only new ones
+	if !config.ExportAll {
+		cmd_list = append(cmd_list, "--state", "new")
+	}
+	err := executeCredentialDiggerProcess(service, cmd_list)
+	if err != nil {
+		log.Entry().Error("failed running credentialdigger get_discoveries")
+		log.Entry().Error(err)
+		return err
+	}
+	log.Entry().Info("Scan complete")
+	return nil
+}
+
+func credentialdiggerBuildCommonArgs(config *credentialdiggerScanOptions) []string {
+	/*Some arguments are the same for all the scan flavors. Build them here
+	* not to duplicate code.*/
+	scan_args := []string{}
+	// Repository url and sqlite db (always mandatory)
+	scan_args = append(scan_args, config.Repository, "--sqlite", piperDbName)
+	//git token is not mandatory for base credential digger tool, but in
+	//piper it is
+	scan_args = append(scan_args, "--git_token", config.Token)
+	//debug
+	if config.Debug {
+		log.Entry().Debug("Run the scan in debug mode")
+		scan_args = append(scan_args, "--debug")
+	}
+	//models
+	if len(config.Models) > 0 {
+		log.Entry().Debugf("Enable models %v", config.Models)
+		scan_args = append(scan_args, "--models")
+		scan_args = append(scan_args, config.Models...)
+	}
+
+	return scan_args
+}
+
+func credentialdiggerScanSnapshot(config *credentialdiggerScanOptions, telemetryData *telemetry.CustomData, service credentialdiggerUtils) error {
+	log.Entry().Infof("Scan Snapshot %v from repo %v", config.Snapshot, config.Repository)
+	cmd_list := []string{"scan_snapshot",
+		"--snapshot", config.Snapshot}
+	cmd_list = append(cmd_list, credentialdiggerBuildCommonArgs(config)...)
+	leaks := executeCredentialDiggerProcess(service, cmd_list)
+	if leaks != nil {
+		log.Entry().Warn("The scan found potential leaks in this Snapshot")
+		return leaks
+	} else {
+		log.Entry().Info("No leaks found")
+		return nil
+	}
+}
+
+func credentialdiggerScanPR(config *credentialdiggerScanOptions, telemetryData *telemetry.CustomData, service credentialdiggerUtils) error {
+	log.Entry().Infof("Scan PR %v from repo %v", config.PrNumber, config.Repository)
+	cmd_list := []string{"scan_pr",
+		"--pr", strconv.Itoa(config.PrNumber),
+		"--api_endpoint", config.APIURL}
```

Comment:
As it is a step working strictly with GitHub, I followed the metadata definition of the GitHub steps, that's why this parameter is called `apiUrl` ([definition here](https://github.com/marcorosa/jenkins-library/blob/main/resources/metadata/credentialdiggerScan.yaml#L50)). 
Besides, the `serverUrl` is not needed for this step.

`api_endpoint` is the naming convention in credentialdigger library, but it will be transparent to piper users.